### ISSUE TITLE
@types/styleslint does not allow extends in LinterOptions

### ISSUE THREAD

**Issue by otakustay** [stylelint/stylelint#3701]
Currently [@types/stylelint](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/stylelint/index.d.ts#L10-L27)'s `LinterOptions` does not allow `extends` key, so when we define a config like:

```typescript
import {LinterOptions} from 'stylelint';

const config: LinterOptions = {
    extends: require.resolve('@ecomfe/stylelint-config')
};

export default config;
```

It reports an error:

```
Type '{ extends: string; }' is not assignable to type 'LinterOptions'.
  Object literal may only specify known properties, and 'extends' does not exist in type 'LinterOptions'.
```

**Comment by otakustay** [comment]
Current types definition seems very different from what stylelint actually exports, the `LinterOptions`, `LintResult` are all incompatible to current implement

**Comment by ybiquitous** [comment]
@otakustay Do you want this `type stylelint$config` actually? 👇 

https://github.com/stylelint/stylelint/blob/a70058f08d5fe1ded44d333a10486e06a05adc54/flow-typed/stylelint.js#L13

https://github.com/stylelint/stylelint/blob/a70058f08d5fe1ded44d333a10486e06a05adc54/flow-typed/stylelint.js#L136

It seems that a same type doesn't be defined in `@types/stylelint`.

```ts
export interface LinterOptions {
  // ...
  config?: JSON;
  // ...
}
```

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fa5666117a2d80fdde2ec29ddd7590a8a7d4af18/types/stylelint/index.d.ts#L13

**Comment by jeddy3** [comment]
Closing for now as this seems related to the `@types/stylelint` package. Feel free to reopen if this isn't the case.
